app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragment.java

```diff
@@ -254,13 +333,32 @@ private boolean requestPermission(){
         return (checkSelfPermission(getContext(), ClassConstants.FINE_LOCATION) == PackageManager.PERMISSION_GRANTED);
     }
 
+    @Override
+    public void onFloorPickerOnClick(int position, View view) {
+        if (selectedFloor != null) selectedFloor.setEnabled(true);
+        selectedFloor = (Button)view;
+        view.setEnabled(false);
+        switch(currentFloorPickerAdapter.getBuildingCode()){
```

Comment:
I believe the same comment I made below applies here as well, why not have a single parametrized method that sets the floor plan according to the parameter passed? This will avoid us having to rewrite source code every time we want to add a floor/new building.